Makefile

```diff
@@ -1,41 +1,21 @@
-# store the current working directory
-CWD := $(shell pwd)
-BASEDIR := $(CWD)
-PRINT_STATUS = export EC=$$?; cd $(CWD); if [ "$$EC" -eq "0" ]; then printf "SUCCESS!\n"; else exit $$EC; fi
+NAME=tendrl-node-agent
 VERSION=0.0.1
 
-BUILDS    := .build
-DEPLOY    := $(BUILDS)/deploy
-PKGNAME   := tendrl-node-agent
-TARDIR    := ${PKGNAME}-$(VERSION)
-RPMBUILD  := $(HOME)/rpmbuild
+all: srpm
 
+clean:
+	rm -rf dist/
+	rm -rf $(NAME)-$(VERSION).tar.gz
+	rm -rf $(NAME)-$(VERSION)-1.el7.src.rpm
 
 dist:
-	rm -fr $(HOME)/$(BUILDS)
-	mkdir -p $(HOME)/$(BUILDS) $(RPMBUILD)/SOURCES
-	cp -fr $(BASEDIR) $(HOME)/$(BUILDS)/$(TARDIR)
-	rm -rf %{pkg_name}.egg-info
-	cd $(HOME)/$(BUILDS); \
-	tar --exclude-vcs --exclude=.* -zcf tendrl-node-agent-$(VERSION).tar.gz $(TARDIR); \
-	cp tendrl-node-agent-$(VERSION).tar.gz $(RPMBUILD)/SOURCES
-        # Cleaning the work directory
-	rm -fr $(HOME)/$(BUILDS)
+	python setup.py sdist \
+	  && mv dist/$(NAME)-$(VERSION).tar.gz .
 
+srpm: dist
+	fedpkg --dist epel7 srpm
 
-rpm:
-	@echo "target: rpm"
-	@echo  "  ...building rpm $(V_ARCH)..."
-	rm -fr $(BUILDS)
-	mkdir -p $(DEPLOY)/latest
-	mkdir -p $(RPMBUILD)/SPECS
-	sed -e "s/@VERSION@/$(VERSION)/" node_agent.spec \
-	        > $(RPMBUILD)/SPECS/node_agent.spec
-	rpmbuild -ba $(RPMBUILD)/SPECS/node_agent.spec
-	$(PRINT_STATUS); \
-	if [ "$$EC" -eq "0" ]; then \
-		FILE=$$(readlink -f $$(find $(RPMBUILD)/RPMS -name tendrl-node-agent-$(VERSION)*.rpm)); \
-		cp -f $$FILE $(DEPLOY)/latest/; \
-		printf "\nThe bridge common RPMs are located at:\n\n"; \
-		printf "   $(DEPLOY)/latest\n\n\n\n"; \
-	fi
+rpm: dist
+	mock -r epel-7-x86_64 rebuild $(NAME)-$(VERSION)-1.el7.src.rpm --resultdir=.
```

Comment:
I recommend adding `--define "dist .el7"` to this mock invocation here, so that it does not use [CentOS's default](http://bugs.centos.org/view.php?id=7416) of `.el7.centos`